src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java

```diff
@@ -1,31 +1,96 @@
 package com.endava.internship.mocking.service;
 
+import com.endava.internship.mocking.model.Payment;
+import com.endava.internship.mocking.model.Status;
+import com.endava.internship.mocking.model.User;
+import com.endava.internship.mocking.repository.PaymentRepository;
+import com.endava.internship.mocking.repository.UserRepository;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.util.*;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
+
+
 @ExtendWith(MockitoExtension.class)
 class PaymentServiceTest {
 
+    private PaymentService service;
+    private UserRepository userRepository;
+    private PaymentRepository paymentRepository;
+    private ValidationService validationService;
+
+    @Captor
+    ArgumentCaptor<Payment> paymentArgumentCaptor;
+
 
     @BeforeEach
     void setUp() {
+        userRepository = mock(UserRepository.class);
+        paymentRepository = mock(PaymentRepository.class);
+        validationService = mock(ValidationService.class);
+        service = new PaymentService(userRepository,paymentRepository,validationService);
+
+
 
     }
 
     @Test
     void createPayment() {
+        User user = new User(1,"User",Status.ACTIVE);
+
+        when(userRepository.findById(1)).thenReturn(Optional.of(user));
+        service.createPayment(user.getId(),200.00);
+
+        verify(validationService).validateUserId(any());
+        verify(validationService).validateAmount(anyDouble());
+        verify(validationService).validateUser(any());
+        verify(userRepository).findById(anyInt());
+
+        paymentArgumentCaptor = ArgumentCaptor.forClass(Payment.class);
+        verify(paymentRepository).save(paymentArgumentCaptor.capture());
+
+        Payment payment = paymentArgumentCaptor.getValue();
+
+        assertThat(user.getId()).isEqualTo(payment.getUserId());
+        assertThat(payment.getMessage()).isEqualTo("Payment from user " + user.getName());
+        assertThat(payment.getAmount()).isEqualTo(200);
 
     }
 
+    @Test
+    void createPaymentThrowExceptionIfUserIsNotFound(){
+        when(userRepository.findById(1)).thenThrow(new NoSuchElementException());
+        Assertions.assertThrows(NoSuchElementException.class,()->service.createPayment(1,200.00));
+
+    }
     @Test
     void editMessage() {
```

Comment:
Could you please check actual payment object with expected one?
We can have one expected payment object and one which is returned from feature.
E.g:
String expectedName = "John";
String actualName = service.getName();
Assertions.assertThat(actualName).isEquals(expected);